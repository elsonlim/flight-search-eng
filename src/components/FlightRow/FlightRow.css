.flights {
    margin-bottom: 10px;
    padding: 15px;
    border: 1px solid #e9e9e9;
}