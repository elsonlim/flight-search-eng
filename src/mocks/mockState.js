import moment from 'moment';

export const flights = [
    {
        "id": 0,
        "IATA": "DL",
        "flightNo": 90,
        "from": "USA",
        "to": "HKG",
        "price": {"amount": 261, "currency": "USD"},
        "time": {"arrival": 609, "departure": 425, "duration": 104}
    }, {
        "id": 1,
        "IATA": "DL",
        "flightNo": 91,
        "from": "HKG",
        "to": "USA",
        "price": {"amount": 232, "currency": "USD"},
        "time": {"arrival": 1956, "departure": 1730, "duration": 146}
    },
];

export const searchParams = {
    isReturn: true,
    from: 'USA',
    to: 'HKG',
    depatureDate: moment(new Date()),
    arrivalDate: moment(new Date()),
    numberOfPassengers: 1,
}

export const priceRange = {
    filter: {min: 0, max: 2000},
    search: {min: 200, max: 1000},
}

export default {
    flights,
    searchType: false,
    searchParams,
    priceRange
};

    