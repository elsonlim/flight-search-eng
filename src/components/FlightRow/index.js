import React, { Component } from 'react';
import { Grid, Button } from 'semantic-ui-react'; 
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import FlightDetail from '../FlightDetail';

import './FlightRow.css';

class FlightRow extends Component {
    getFlightDetails() {
        const { departure, arrival } = this.props;

        if (arrival) {
            return (
                <div>
                    <Grid columns={'equal'}>
                        <Grid.Column>
                            <FlightDetail flightInfo={departure} />
                        </Grid.Column>
                        <Grid.Column>
                            <FlightDetail flightInfo={arrival} />
                        </Grid.Column>
                    </Grid>
                </div>
            );
        }

        return (<FlightDetail flightInfo={departure} />);
    }
    
    render() {
        const { price } = this.props;

        return (
            <div className="flights">
                <Grid>
                    <Grid.Row verticalAlign={'middle'}>
                        <Grid.Column width={10}>
                            {this.getFlightDetails()}
                        </Grid.Column>
                        <Grid.Column width={2}>
                            ${price}
                        </Grid.Column>
                        <Grid.Column width={4}>
                            <Button content='Select flight' primary /> 
                        </Grid.Column>
                    </Grid.Row>
                </Grid>
            </div>    
        );
    }
}

const mapStateToProps = state => ({
    numberOfPassengers: state.searchParams.numberOfPassengers
});

FlightRow.PropTypes = {
    numberOfPassengers: PropTypes.number.isRequired,
    departure: PropTypes.object.isRequired,
    arrival: PropTypes.object.isRequired,
};

export default connect(mapStateToProps)(FlightRow);