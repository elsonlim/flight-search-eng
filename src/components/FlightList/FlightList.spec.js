import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import enzyme, { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
import moment from 'moment';
import configureStore from 'redux-mock-store';
import {FlightList} from './index';
import FlightRow from '../FlightRow';
import mockState from '../../mocks/mockState';

enzyme.configure({ adapter: new Adapter() });

describe('FlightList', () => {

    let props;
    let wrapper;
    let store;
    const mockStore = configureStore();

    beforeEach(() => {
        props = mockState;
        store =  mockStore(mockState);
        wrapper = shallow(<FlightList {...props} />);
    });
    
    describe('snapshot test', () => {
		test('show flightlist', () => {
			const tree = renderer.create(<Provider store={store}><FlightList {...props} /></Provider>).toJSON();
			expect(tree).toMatchSnapshot();
		});
    });

    describe('_getFilteredFlights', () => {
        test('should have 1 departing and 1 arriving flight', () => {
            const flights = wrapper.instance()._getFilteredFlights();
    
            expect(flights.arriving).toHaveLength(1);
            expect(flights.departing).toHaveLength(1);
        })
    })

    describe('_showOnewayFlights', () => {
        test('should have 1 flightrow', () => {
            const flights = wrapper.instance()._getFilteredFlights();
            const filteredFlights = wrapper.instance()._showOnewayFlights(flights);

            expect(filteredFlights).toHaveLength(1);
        });
    });

    describe('_showReturnFlights', () => {
        test('should have 1 flightrow', () => {
            const flights = wrapper.instance()._getFilteredFlights();
            const filteredFlights = wrapper.instance()._showReturnFlights(flights);

            expect(filteredFlights).toHaveLength(1);
        });
    });

    describe('_checkRefineSearch', () => {
        test('return true if price within range', () => {
            const withinRange = wrapper.instance()._checkRefineSearch(500);

            expect(withinRange).toBeTruthy();
        });

        test('return false if price above range', () => {
            const withinRange = wrapper.instance()._checkRefineSearch(1500);

            expect(withinRange).toBeFalsy();
        });
    });

    describe('getFlightDate', () => {
        test('show 1 date field if is not return flight', () => {
            expect(wrapper.instance().getFlightDate(false)).toHaveLength(1);
        });

        test('show 2 date field if is not return flight', () => {
            expect(wrapper.instance().getFlightDate(true)).toHaveLength(2);
        });
    });
});
