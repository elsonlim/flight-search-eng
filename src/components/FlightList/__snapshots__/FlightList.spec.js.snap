// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightList snapshot test show flightlist 1`] = `
<div>
  <div
    className="results-header-container"
  >
    <div
      className="results-title"
    >
      Your Results
    </div>
    <div
      className="results-date"
    >
      <div>
        Depart: 
        2018-5-20
      </div>
      <div>
        Return: 
        2018-5-20
      </div>
    </div>
  </div>
  <div
    className="flights"
  >
    <div
      className="ui grid"
    >
      <div
        className="middle aligned row"
      >
        <div
          className="ten wide column"
        >
          <div>
            <div
              className="ui equal width grid"
            >
              <div
                className="column"
              >
                <div
                  className="flight-details"
                >
                  <div
                    className="logo"
                  >
                    <div>
                      <img
                        className="ui tiny rounded image"
                        src="/airlineLogo/dl.svg"
                      />
                    </div>
                  </div>
                  <div
                    className="details"
                  >
                    <div>
                      DL
                      -
                      90
                    </div>
                    <div>
                      USA
                       &gt; 
                      HKG
                    </div>
                    <div>
                      Depart: 
                      04:25am
                    </div>
                    <div>
                      Arrival: 
                      06:09am
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="column"
              >
                <div
                  className="flight-details"
                >
                  <div
                    className="logo"
                  >
                    <div>
                      <img
                        className="ui tiny rounded image"
                        src="/airlineLogo/dl.svg"
                      />
                    </div>
                  </div>
                  <div
                    className="details"
                  >
                    <div>
                      DL
                      -
                      91
                    </div>
                    <div>
                      HKG
                       &gt; 
                      USA
                    </div>
                    <div>
                      Depart: 
                      05:30pm
                    </div>
                    <div>
                      Arrival: 
                      07:56pm
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="two wide column"
        >
          $
          493
        </div>
        <div
          className="four wide column"
        >
          <button
            className="ui primary button"
            disabled={undefined}
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
          >
            Select flight
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
